File: leetcode/editor/cn/257_二叉树的所有路径_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"strconv"
	"testing"
)

func Test_givenNormal_when257_thenSuccess(t *testing.T) {
	assert.Equal(t, 1, 1, "shouldEqual")
}

//leetcode submit region begin(Prohibit modification and deletion)
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func binaryTreePaths(root *TreeNode) []string {
	var ans []string
	if root == nil {
		return ans
	}
	rootVal := strconv.Itoa(root.Val)
	if root.Left != nil {
		for _, item := range binaryTreePaths(root.Left) {
			ans = append(ans, rootVal+"->"+item)
		}
	}
	if root.Right != nil {
		for _, item := range binaryTreePaths(root.Right) {
			ans = append(ans, rootVal+"->"+item)
		}
	}
	if root.Left == nil && root.Right == nil {
		ans = append(ans, rootVal)
	}
	return ans
}

//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/cn/257_二叉树的所有路径_test.py
import unittest
from typing import Optional, List

from model.tree_node import TreeNode


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        root = TreeNode.decode([1, 2, 3, None, 5])
        paths = self.solution.binaryTreePaths(root)
        self.assertListEqual(["1->2->5", "1->3"], paths)


# leetcode submit region begin(Prohibit modification and deletion)
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        ans = []
        if root is None:
            return ans
        if root.left:
            for item in self.binaryTreePaths(root.left):
                ans.append(str(root.val) + "->" + item)
        if root.right:
            for item in self.binaryTreePaths(root.right):
                ans.append(str(root.val) + "->" + item)
        if root.left is None and root.right is None:
            ans.append(str(root.val))
        return ans
        pass


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()
